### ISSUE TITLE
NPC wolf does not have get red eyes (angry skin) after command "/npc wolf -a"

### ISSUE THREAD

**Issue by Qveshn** [CitizensDev/Citizens2#1452]
I can not make npc with type=wolf angry :(
spigot 1.12.2
citizens 2.0.22 build #1524
Command: /npc wolf -a
This command set/clear property "angry" in the file saves.yml. But no rendering. 
More precisely: client renders npc as angry wolf for a few microseconds and then his eyes (with skin) back to normal wolf skin.
No errors in console.

**Comment by mcmonkey4eva** [comment]
Possible some internal (Spigot or Minecraft) logic is removing the effect. We may need to add some workaround code to Citizens for that, if it replicates consistently without other plugins involved.

**Comment by Qveshn** [comment]
I have done all tests on spigot with only one plugin citizens.

**Comment by Qveshn** [comment]
I found error =)
net.citizensnpcs.nms.v1_10_R1.entity.WolfController.EntityWolfNPC#setGoalTarget
net.citizensnpcs.nms.v1_11_R1.entity.WolfController.EntityWolfNPC#setGoalTarget
net.citizensnpcs.nms.v1_12_R1.entity.WolfController.EntityWolfNPC#setGoalTarget

You should change == to !=
Current:
```
        @Override
        public boolean setGoalTarget(EntityLiving entityliving, EntityTargetEvent.TargetReason reason, boolean fire) {
            return npc == null ? super.setGoalTarget(entityliving, reason, fire) : false;
        }
```
Must be:
```
        @Override
        public boolean setGoalTarget(EntityLiving entityliving, EntityTargetEvent.TargetReason reason, boolean fire) {
            return npc != null ? super.setGoalTarget(entityliving, reason, fire) : false;
        }
```
:stuck_out_tongue_winking_eye:

**Comment by Qveshn** [comment]
Wait a moment...  Something go wrong... I need more tests :worried:

**Comment by Qveshn** [comment]
I have looked closely to code :blush:
Now I understood what all that meant
You have WolfController to intercept creation wolf as citizen. But spigot also do creation his wolves over your controller. So, the main task with next code was to not allow npc-wolf to be angry while somebody hit normal wolf nearby:
```
        @Override
        public boolean setGoalTarget(EntityLiving entityliving, EntityTargetEvent.TargetReason reason, boolean fire) {
            return npc == null ? super.setGoalTarget(entityliving, reason, fire) : false;
        }
```
But problem was when command "/npc wolf -a" call wolf.setTarget(wolf); in method net.citizensnpcs.trait.WolfModifiers#updateModifiers. Spigot calls method setGoalTarget and code above prevent our npc to set target (npc!=null). So, we could not set wolf angry.
I lool at code `wolf.setTarget(wolf);` and noticed that npc-wolf (when angry) setd target to himself. So, I have used it to workaround problem in method setGoalTarget:
```
        @Override
        public boolean setGoalTarget(EntityLiving entityliving, EntityTargetEvent.TargetReason reason, boolean fire) {
            return npc == null || this == entityliving ? super.setGoalTarget(entityliving, reason, fire) : false;
        }
```
This code must be changed in all nms versions:
net.citizensnpcs.nms.v1_10_R1.entity.WolfController.EntityWolfNPC#setGoalTarget
net.citizensnpcs.nms.v1_11_R1.entity.WolfController.EntityWolfNPC#setGoalTarget
net.citizensnpcs.nms.v1_12_R1.entity.WolfController.EntityWolfNPC#setGoalTarget

`this==entityliving` means that we also allow to set target if entiy target is himself.
Also, we can use `this.equals(entityliving)` that compers entity Id, but I think that compare refernces is also good.
:blush:
